### ISSUE TITLE
Something went wrong fetching user data: 401

### ISSUE THREAD

**Issue by mjkl-gh** [pwesters/watts_vision#2]
Unfortunately lately i'm getting a 401 error with my credentials. I've retried setting it up multiple times and I've verified my e-mailadres and password against the one I'm filling in on the web app (https://smarthome.wattselectronics.com/) and there they work.

DId something change?
```
2022-07-04 20:12:29 ERROR (SyncWorker_4) [custom_components.watts_vision.watts_api] Something went wrong fetching user data: 401
```
```
2022-07-04 20:15:02 ERROR (MainThread) [homeassistant.components.climate] Error while setting up watts_vision platform for climate
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 249, in _async_setup_platform
await asyncio.shield(task)
File "/config/custom_components/watts_vision/climate.py", line 59, in async_setup_entry
for y in range(len(smartHomes)):
TypeError: object of type 'NoneType' has no len()
```

**Comment by pwesters** [comment]
Yes, something changed, I do not yet know what it is. I have to look into it.

**Comment by huijsh** [comment]
got the same error. Any idea yet ?

**Comment by mjkl-gh** [comment]
I've been able to fix the 401 error. When disecting the web app I noticed it now also sends the token in the payload. Moreover, if the token is expired it just returns 401. This isn't handled correctly in the current code.

Moreover, the result now returns Code: 8, Key: OK_SET, Value: Insert / update success instead of 1, OK, OK 

It does this both in the web app as in the python code.  If you fix the result check it now fails something else. Probably the result itself was changed

Long story short, they changed the API significantly. I will report back when I figure something working out.

**Comment by mjkl-gh** [comment]
@pwesters is there a specific reason for doing: 

```
_smartHomeData[str(y)] 
```

Currently it's erroring out everywhere because it won't accept str for an index. I'm not sure what I changed to cause that.

However, To me it seems more logical to change it to:
```
for data in self._smartHomeData:
    devices = self.loadDevices(data["smarthome_id"])
    data["devices"] = devices
```

**Comment by pwesters** [comment]
I finally got some time to look into it. The api changed. The api is giving a proper 401 when authorization fails, it is using arrays instead of objects when appropriate and the codes are changed. That's what I found at the moment.

I will try to fix it as soon as possible.

**Comment by mjkl-gh** [comment]
> I finally got some time to look into it. The api changed. The api is giving a proper 401 when authorization fails, it is using arrays instead of objects when appropriate and the codes are changed. That's what I found at the moment.
> 
> I will try to fix it as soon as possible.

I've already implemented some dirty fixes at my [fork](https://github.com/mjkl-gh/watts_vision). Now that I know the [str(y)] was because of the switch to arrays instead of objects. I can make a dirty fix for those so we get something working in the meantime.

After that. Maybe we can combine efforts. I've started work on seperating the Watts api into a seperate async pypi package at https://github.com/mjkl-gh/aio-watts-vision
As prescribed by home assistant at https://developers.home-assistant.io/docs/api_lib_index/

I'd also like to properly incorperate the openid-connect procedures. Right now the tokens are refreshed after they become invalid. However, with token a refresh_token is sent and an expiration date. We can just check if the received token has expired and use the refresh_token to request a new one. 

I've seen the web app actually use this  procedure and there are a ton of examples in other home assistant integrations that incorporate this mechanism, like pyoverkiz.

**Comment by pwesters** [comment]
Sounds good. The refresh token I already tried once but still got token expired issues. If you managed to get it working or have a working example I would like to know.

I created a new release that should address the changes in the Watts Vision API.
